apps/web/src/pages/get-started/components/OpenWorkflowButton.tsx

```diff
@@ -1,14 +1,28 @@
+import { errorMessage } from '@novu/design-system';
 import { INotificationTemplate } from '@novu/shared';
 
-import { ROUTES } from '../../../constants/routes.enum';
-import { useCreateTemplateFromBlueprint } from '../../../api/hooks';
-import { errorMessage } from '../../../utils/notifications';
-import { getBlueprintTemplateById, getTemplateById } from '../../../api/notification-templates';
-import { getWorkflowBlueprintDetails } from '../../../utils';
 import { TemplateCreationSourceEnum } from '../../templates/shared';
 import { useSegment } from '../../../components/providers/SegmentProvider';
 import { OnboardingWorkflowRouteEnum } from '../consts/types';
 import { StyledLink } from '../consts/shared';
+import { useCreateWorkflowFromBlueprint } from '../../../hooks';
+import { openInNewTab } from '../../../utils';
+import { parseUrl } from '../../../utils/routeUtils';
+import { ROUTES } from '../../../constants/routes.enum';
+
+function buildWorkflowEditorUrl(workflow: INotificationTemplate, step?: OnboardingWorkflowRouteEnum) {
+  const workflowRoute = parseUrl(ROUTES.WORKFLOWS_EDIT_TEMPLATEID, { templateId: workflow._id! });
+  if (step === OnboardingWorkflowRouteEnum.TEST_WORKFLOW) {
+    return `${window.location.origin}${workflowRoute}/${OnboardingWorkflowRouteEnum.TEST_WORKFLOW}`;
+  }
+
+  const stepUuid = workflow.steps.find((el) => el.name?.toLowerCase() === step?.toLowerCase())?.uuid;
+  if (step && stepUuid) {
+    return `${window.location.origin}${workflowRoute}/${step}/${stepUuid}`;
+  }
+
+  return `${window.location.origin}${workflowRoute}`;
+}
```

Comment:
This function builds the URL to the Workflow Editor. When:
- step is TEST_WORKFLOW the URL point to the Test Workflow page with sidebar opened: `http://localhost:4200/workflows/edit/65d91b93b0bb8c7b5e68a9ab/test-workflow`
- if the workflow has the step redirect him to the step editor: `http://localhost:4200/workflows/edit/65d91b93b0bb8c7b5e68a9ab/in-app/65d91b93b0bb8c7b5e68a9ab`
- otherwise redirect to the Workflow Editor: `http://localhost:4200/workflows/edit/65d91b93b0bb8c7b5e68a9ab`